libr/util/vector.c

```diff
@@ -19,10 +19,15 @@
 
 #define RESIZE_OR_RETURN_NULL(next_capacity) do { \
 		size_t new_capacity = next_capacity; \
-		void **new_a = realloc (vec->a, vec->elem_size * new_capacity); \
+		size_t new_capacity_size = vec->elem_size * new_capacity; \
+		void **new_a = realloc (vec->a, new_capacity_size); \
 		if (!new_a) { \
 			return NULL; \
 		} \
+		if (vec->capacity < new_capacity) { \
+			size_t old_capacity_size = vec->elem_size * vec->capacity; \
+			memset(&new_a[vec->capacity], 0, new_capacity_size - old_capacity_size); \
```

Comment:
Add a space before (. A review from @thestr4ng3r may be good here

Reply:
why is this necessary @wargio ? AFAIK you should not access the elements after `vec->len`, so it doesn't really matter what is the value inside those elements.

Reply:
ok, the binary i have does something that should not happen. essentially resizes the vector (old < new) and then fails to add stuff. the `realloc` does not clean the memory when you resize with a bigger chunk of memory and the vector free fails to free garbage pointer (obviously).